src/script/script.js

```diff
@@ -536,6 +536,28 @@ pc.extend(pc, function () {
                     this.fire(this.enabled ? 'enable' : 'disable');
                     this.fire('state', this.enabled);
                 }
+
+                // initialize script if not initialized yet and script is enabled
+                if (! this._initialized && this.enabled) {
```

Comment:
I think that it is better to check `this.enable` only once.

```js
if (!this.enabled) return;
```
or
```js
if (this.enable) {
  if (! this._initialized) {
    ...
  }
  if (! this._postInitialized && !this.entity.script._beingEnabled) {
    ...
  }
}
```

Reply:
The thing is that when you execute the initialize method then the script might no longer be enabled so we need to check `this.enabled` again.

Reply:
Oh...  :disappointed:

Reply:
if this.enable is flip-flopped, I think that should call fire method after here.